### PR TITLE
#161 - Added BasicAuthorizationHeader to HTTP module

### LINKED ISSUE #161: Add `BasicAuthorizationHeader`
It would be very helpful to have a way to create `Authorization` header for Basic auth in tests.

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #161

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#162](https://github.com/artipie/http/issues/162) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/9b886d2b-3e4c-4391-9de9-4dec96217d31); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/http/auth/BasicAuthorizationHeader.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.artipie.http.rs.Header;
+import org.cactoos.text.Base64Encoded;
+
+/**
+ * Basic authentication `Authorization` header.
+ *
+ * @since 0.9.5
+ */
+public class BasicAuthorizationHeader extends Header.Wrap {
```
Comment (carlosmiranda) [inline src/main/java/com/artipie/http/auth/BasicAuthorizationHeader.java:34]: @olegmoz why not make this class final?

File: src/main/java/com/artipie/http/auth/BasicAuthorizationHeader.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * Ctor.
+     *
+     * @param username User name.
+     * @param password Password.
+     */
+    protected BasicAuthorizationHeader(final String username, final String password) {
```
Comment (carlosmiranda) [inline src/main/java/com/artipie/http/auth/BasicAuthorizationHeader.java:42]: @olegmoz likewise, why is this protected? Are we expecting to extend this class to use this constructor?

**Comment by olegmoz** [comment]
@carlosmiranda thank you for review, fixed both points.

**Comment by 0crat** [comment]
Job was finished in 24 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@carlosmiranda please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@carlosmiranda ping

**Comment by carlosmiranda** [comment]
@sereshqua confirmed, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**